### ISSUE TITLE
after browser restart, completed downloads shown as "missing"

### ISSUE THREAD

**Issue by stevepatter** [downthemall/downthemall#65]
**Desktop (please complete the following information):**
 - OS: [e.g. Linux]
 - Browser [e.g. Firefox]
 - Version [e.g. 69]

**Describe the bug**
If I successfully download 1 or more files with DTA & then restart Firefox, the previously completed downloads are shown in the download list as "missing" with a red progress bar & delete icon.  

As these downloads have already completed, I think they should instead show as "complete"

**Comment by nmaier** [comment]
Well, they are missing from the download history as both Firefox and chrome move them into history where we got no way of accessing anymore.

DownThemAll! has an option to clear such downloads automatically, but that's really the best we can do here.

**Comment by aletaziar** [comment]
While DTA still has access to the downloads, would it be possible to copy the file path to an internal DTA database so finished files are at least still openable?

**Comment by nmaier** [comment]
>While DTA still has access to the downloads, would it be possible to copy the file path to an internal DTA database so finished files are at least still openable?

No, DownThemAll! (and other WebExtensions for that matter) can only access files through the downloads API, and only then if there is still a valid download "item" that it can query. Once the items get moved to history by the browser, WebExtensions lose access to these download items.

Keeping a backup record wouldn't help, as WebExtensions still couldn't do anything with the files.

This is in my opinion a clear misfeature in the browsers, but it is what it is.
